lib/housing-available/housing-available.controllers.js

```diff
@@ -1 +1,229 @@
+  // const housingAvailableRoutes = require('./housing-available.routes');
+  const Log = require('../log/log.controllers')(models).lib;
+  const accountLib = require('../account/account.controllers')(models).lib;
+  const propertyLib = require('../property/property.controllers')(models).lib;
+  const locationLib = require('../location/location.controllers')(models).lib;
+  const postalAddressLib = require("../postal-address/postal-address.controllers")(models).lib;
+
+  return {
+
+    // Create new housingAvailable rental listing;
+    postHousingAvailable: async function (request, h) {
+      /*
+        # Require Account; user must be a logged-in user (i.e. have an Account) to post a rental;
+      */
+      let account;
+      let payload = request.payload;
+      try {
+        account = await accountLib.getAccount(request.auth.credentials);
+      } catch (e) {
+        throw Boom.badImplementation('Error during getAccount(request.auth.credentials). ' + e);
+      }
+
+      if (account === null) {
+        throw Boom.badRequest('Must have an Account to create a HousingAvailable');
+      }
+
+      /* Restrict AuthorId to same user, or SuperAdmin */
+      if (payload.AuthorId) {
+        // If AuthorId is set then we have already passed auth and account lookup
+        // This is because if AuthorId is set then not only quickInfo is set
+        if (
+          payload.AuthorId !== account.id &&
+          request.auth.credentials.subjectId !== process.env.SUPER_ADMIN
+        ) {
+          throw Boom.badRequest('HousingAvailable.AuthorId must match your Account ID or you must be a Super Admin');
+        }
+      }
+
+      // bedrooms validation:
+      let hasBedrooms = false;
+      let bedrooms = Number(request.payload.bedrooms);
+      if (!ensureNonnegativeValue(bedrooms)) {
+        throw Boom.badData("Bedrooms cannot be a negative number.")
+      }
+      if (request.payload.hasOwnProperty("bedrooms")) {
+        hasBedrooms = true;
+      }
+
+      // bathrooms validation:
+      let hasBathrooms = false;
+      let bathrooms = Number(request.payload.bathrooms);
+      if (!ensureNonnegativeValue(bathrooms)) {
+        throw Boom.badData("Bathrooms cannot be a negative number.")
+      }
+      if (request.payload.hasOwnProperty("bathrooms")) {
+        hasBathrooms = true;
+      }
+
+      // bathrooms can be in .25 increments
+      if (hasBathrooms && !bathroomsValidIncrement(bathrooms)) {
+        throw Boom.badData('bathrooms must be in .25 increments');
+      }
+
+      // start building HousingAvailable object
+      let housingAvailableObject = {};
+      if (request.payload.title && request.payload.title !== request.payload.address) {
+        housingAvailableObject.title = request.payload.title;
+      }
+
+      // This sets the AuthorId associated with this HA listing to === AuthorId of the post itself;
+      if (request.payload.AuthorId) {
```

Comment:
there should be an else here that sets AuthorId to account.id if AuthorId isn't in the payload (because it's not required). In the case of HousingAvailable, we never want the content to be un-owned.